.taskcluster.yml

```diff
@@ -6,259 +6,212 @@ version: 1
+            MOBILE_HEAD_BRANCH: ${head_branch}
+            MOBILE_HEAD_REV: ${head_rev}
           features:
             taskclusterProxy: true
-            chainOfTrust: true
-          artifacts:
-            public/task-graph.json:
-              type: file
-              path: /build/android-components/task-graph.json
-              expires: ${expires_in}
-            public/actions.json:
-              type: file
-              path: /build/android-components/actions.json
-              expires: ${expires_in}
-            public/parameters.yml:
-              type: file
-              path: /build/android-components/parameters.yml
-              expires: ${expires_in}
         extra:
           tasks_for: ${tasks_for}
         metadata:
-          name: Android Components - Decision task (${tag})
-          description: Build and publish release versions.
           owner: ${user}@users.noreply.github.com
-          source: ${repository}/raw/${tag}/.taskcluster.yml
+          source: ${repository}/raw/${head_rev}/.taskcluster.yml
+    in:
+      $if: 'tasks_for in ["github-pull-request", "github-push"]'
+      then:
+        $let:
+          pr_or_push_parameters:
+            scopes:
+              - queue:create-task:aws-provisioner-v1/github-worker
+              - secrets:get:project/mobile/android-components/public-tokens
+            payload:
+              command:
+                - >-
+                  git fetch ${repository} ${head_branch}
+                  && git config advice.detachedHead false
+                  && git checkout ${head_rev}
+                  && ./gradlew --no-daemon --version
+                  && python automation/taskcluster/decision_task.py
+        in:
+          - $if: 'tasks_for == "github-pull-request" && event["action"] in ["opened", "reopened", "edited", "synchronize"]'
+            then:
+              $let:
+                pull_request_title: ${event.pull_request.title}
+                pull_request_number: ${event.pull_request.number}
+                pull_request_url: ${event.pull_request.html_url}
+              in:
+                $mergeDeep:
+                  - {$eval: 'default_task_definition'}
+                  - {$eval: 'pr_or_push_parameters'}
+                  - payload:
+                      env:
+                        GITHUB_PULL_TITLE: ${pull_request_title}
+                    metadata:
+                      name: 'Android Components - Decision task (Pull Request #${pull_request_number})'
+                      description: 'Building and testing Android components - triggered by [#${pull_request_number}](${pull_request_url})'
+          - $if: 'tasks_for == "github-push" && head_branch[:10] != "refs/tags/"'
+            then:
+              $mergeDeep:
+                - {$eval: 'default_task_definition'}
+                - {$eval: 'pr_or_push_parameters'}
+                - metadata:
+                    name: Android Components - Decision task
```

Comment:
Quotes are needed in the `pull_request` section because of the `#` character. If quotes aren't there, it's interpreted as a comment section.

Here, the name is simple enough to not need a pair quote. I'm fine adding them here, but as it's unnecessary, I perceive it as clutter. That said, I know it's my personal taste. If you do want them to be added, I won't argue more. Please let me know what you think!